packages/web/app/src/pages/target-trace.tsx

```diff
@@ -0,0 +1,1784 @@
+                      />
+                    </div>
+                  ) : null}
+                </div>
+              </ScrollArea>
+            </div>
+          </ResizablePanel>
+        </ResizablePanelGroup>
+      </TooltipProvider>
+      {props.activeSpanId && (
+        <SpanSheet
+          span={trace.spans.find(trace => trace.id === props.activeSpanId) ?? null}
+          computedSpanMetrics={spansById.get(props.activeSpanId) ?? null}
+          onClose={() =>
+            navigate({
+              to: '/$organizationSlug/$projectSlug/$targetSlug/trace/$traceId',
+              search: {},
+            })
+          }
+          organizationSlug={props.organizationSlug}
+          projectSlug={props.projectSlug}
+          targetSlug={props.targetSlug}
+          traceId={trace.id}
+          activeTab={props.activeSpanTab}
+        />
+      )}
+    </div>
+  );
+}
+
+const TargetInsightsNewPageContent_TraceQuery = graphql(/* GraphQL */ `
+  query TargetInsightsNewPageContent_TraceQuery(
+    $targetSelector: TargetSelectorInput!
+    $traceId: ID!
+  ) {
+    target(reference: { bySelector: $targetSelector }) {
+      id
+      trace(traceId: $traceId) {
+        ...TraceSheet_TraceFragment
+        id
+        operationName
+        duration
+        success
+        timestamp
+        spans {
+          id
+          name
+          parentId
+          ...SpanFragment
+        }
+      }
+    }
+  }
+`);
+
+function TargetInsightsNewPageContent(props: {
+  organizationSlug: string;
+  projectSlug: string;
+  targetSlug: string;
+  traceId: string;
+  activeSpanId: string | null;
+  activeSpanTab: string | null;
+}) {
+  const [result] = useQuery({
+    query: TargetInsightsNewPageContent_TraceQuery,
+    variables: {
+      targetSelector: {
+        organizationSlug: props.organizationSlug,
+        projectSlug: props.projectSlug,
+        targetSlug: props.targetSlug,
+      },
+      traceId: props.traceId,
+    },
+  });
+
+  const isFetching = result.fetching || result.stale;
+  const trace = result.data?.target?.trace;
+
+  return (
+    <div className="flex h-full flex-col space-y-4 pt-6">
+      <Meta title={`Trace ${props.traceId}`} />
+      <SubPageLayoutHeader
+        subPageTitle={
+          <span className="flex items-center">
+            <Link
+              to="/$organizationSlug/$projectSlug/$targetSlug/traces"
+              params={{
+                organizationSlug: props.organizationSlug,
+                projectSlug: props.projectSlug,
+                targetSlug: props.targetSlug,
+              }}
+            >
+              Traces
+            </Link>{' '}
+            <span className="inline-block px-2 italic text-gray-500">/</span>{' '}
+            {trace ? (
+              <>
+                {trace.operationName ?? <span className="text-gray-400">{'<unknown>'}</span>}
+                <span className="text-muted-foreground ml-2 font-mono font-normal">
```

Comment:
```suggestion
                <span className="text-muted-foreground ml-2 font-mono font-normal" title={trace.id}>
```
a quick way to allow seeing full trace ID on hover